runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java

```diff
@@ -61,65 +125,76 @@ public static StreamAddressSpace merge(StreamAddressSpace a, StreamAddressSpace
         return a;
     }
 
-    // Merge another StreamAddressSpace into this instance
+    /**
+     * Merge another StreamAddressSpace into this instance
+     * @param other the other StreamAddressSpace to merge into this
+     */
     private void mergeFrom(StreamAddressSpace other) {
         this.trimMark = Long.max(trimMark, other.getTrimMark());
-        this.addressMap.or(other.getAddressMap());
+        this.bitmap.or(other.getBitmap());
 
         // Because the trim mark can increase after the merge another
         // trim needs to be issued
         this.trim(this.trimMark);
     }
 
+    /**
+     * Visit all values in the bitmap and pass them to the consumer.
+     * @param longConsumer consumer
+     */
+    public void forEach(final LongConsumer longConsumer) {
+        bitmap.forEach(longConsumer);
+    }
+
     /**
      * Copy this stream's addresses to a set, under a given boundary (inclusive).
      *
      * @param maxGlobal maximum address (inclusive upper bound)
      */
-    public NavigableSet<Long> copyAddressesToSet(final Long maxGlobal) {
-        NavigableSet<Long> queue = new TreeSet<>();
-        this.addressMap.forEach(address -> {
+    public NavigableSet<Long> getAddressesUpTo(final Long maxGlobal) {
+        NavigableSet<Long> subset = new TreeSet<>();
+        bitmap.forEach(address -> {
```

Comment:
Using `Long` instead of `long` is silly, but required to force compilation errors when code tries add `int`, although that should happen, silent failures can happen due to casting.

Hopefully the compiler can recognize that these `Long` objects are not needed and hences replaces with the primitive type.